src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -219,50 +212,46 @@ public static boolean canContinueUsing(@Nonnull ItemStack from, @Nonnull ItemSta
         return false;
     }
 
-    public static boolean canHarvestBlock(@Nonnull BlockState state, @Nonnull Player player, @Nonnull BlockGetter world, @Nonnull BlockPos pos)
+    public static boolean isCorrectToolForDrops(@Nonnull BlockState state, @Nonnull Player player)
     {
-        //state = state.getActualState(world, pos);
         if (!state.requiresCorrectToolForDrops())
             return ForgeEventFactory.doPlayerHarvestCheck(player, state, true);
 
-        ItemStack stack = player.getMainHandItem();
-        ToolType tool = state.getHarvestTool();
-        if (stack.isEmpty() || tool == null)
-            return player.hasCorrectToolForDrops(state);
-
-        int toolLevel = stack.getHarvestLevel(tool, player, state);
-        if (toolLevel < 0)
-            return player.hasCorrectToolForDrops(state);
-
-        return ForgeEventFactory.doPlayerHarvestCheck(player, state, toolLevel >= state.getHarvestLevel());
+        return player.hasCorrectToolForDrops(state);
     }
 
-    public static boolean isToolEffective(LevelReader world, BlockPos pos, @Nonnull ItemStack stack)
+    public static boolean isCorrectToolForDrops(TieredItem tieredItem, ItemStack stack, BlockState state)
```

Comment:
Can this be changed to `IForgeTieredItem` instead of `TieredItem`? Would be easier for extension as you could implement that interface on a different clas.

Reply:
Is there a real use case for having getTier() in a class that isn't TieredItem? If so I'd move the method to IForgeItem so any item can implement it.